classes/class-newsmatchdonation-settings.php

```diff
@@ -0,0 +1,257 @@
+<?php
+/**
+ * The class for the News Match Donation plugin's setting
+ *
+ * @package NewsMatchDonation\Settings
+ */
+
+/**
+ * Register and populate a settings page for the News Match Donation Plugin:
+ */
+class NewsMatchDonation_Settings {
+	/**
+	 * The slug of the settings page
+	 *
+	 * @var string $settings_page The settings page slug
+	 */
+	public $settings_page = 'newsmatchdonation';
+
+	/**
+	 * The slug of the settings group
+	 *
+	 * @var string $settings_group The settings group slug
+	 */
+	public $settings_group = 'newsmatchdonation_group';
+
+	/**
+	 * The slug of the settings section
+	 *
+	 * @var string $settings_section The slug of the settings section
+	 */
+	public $settings_section = 'newsmatchdonation_section';
+
+	/**
+	 * The prefix used for this plugin's options saved in the options table
+	 *
+	 * @var string $options_prefix The prefix for this plugin's options saved in the options table
+	 */
+	public static $options_prefix = 'nmds_';
```

Comment:
what's the purpose of making this variable static?

Reply:
Because it's not going to change during the page load, but it is used by the shortcode class. The shortcode class doesn't need to instantiate a new copy of the settings class; it just needs to pull canonical information from the settings class.

I guess I could pass the settings class into the shortcode's constructor?